src/Renci.SshNet/SshClient.cs

```diff
@@ -450,6 +499,21 @@ public ShellStream CreateShellStream(string terminalName, uint columns, uint row
             return ServiceFactory.CreateShellStream(Session, terminalName, columns, rows, width, height, terminalModeValues, bufferSize);
         }
 
+        /// <summary>
+        /// Creates the shell stream.
+        /// </summary>
+        /// <param name="bufferSize">The size of the buffer.</param>
+        /// <returns>
+        /// The created <see cref="ShellStream"/> instance.
+        /// </returns>
+        /// <exception cref="SshConnectionException">Client is not connected.</exception>
+        public ShellStream CreateShellStream(int bufferSize)
```

Comment:
Let's make it an optional argument so we can give it a default value when it is not supplied (-1 means default)

```suggestion
        public ShellStream CreateShellStream(int bufferSize = -1)
```

Reply:
There are many methods take `bufferSize` argument. Should we change them all?
![image](https://github.com/sshnet/SSH.NET/assets/1913505/348b0b16-9c0f-4dad-b037-b87d8788323c)

Reply:
No I don't think any others are worth changing, just this new API. I'm guessing most of these overloads were added before optional arguments existed

Reply:
Then other methods will throw exception if buffer size is -1, whereas this method will treat -1 as 1024. Is that okay for you?

Reply:
You could add

```c# 
if (bufferSize == -1)
{
    bufferSize = 1024;
}
```

before the `ArgumentOutOfRangeException.ThrowIfNegativeOrZero` in the constructor to make it consistent for all overloads

Reply:
How about making the optional `bufferSize` default to 1024 rather than -1?

Reply:
```C#
Shell CreateShellNoTerminal(Stream input, Stream output, Stream extendedOutput, int bufferSize = 1024)

ShellStream CreateShellStreamNoTerminal(int bufferSize = 1024)
```

Reply:
-1 is better because then we can change the default in the future as an implementation detail. See StreamWriter https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs#L93

Reply:
Ok, that makes sense. Thanks.